### PR TITLE
Add support for merge queue ruleset JSON unmarshaling 

### PR DESCRIPTION
**Pull request by Matthew-Reidy**
Fixes: #3098.

Added support for respository ruleset merge_queue in JSON unmarshalling code with accompanying tests

### CODE REVIEW

File: AUTHORS
Comment (gmlewis) [inline AUTHORS:488]: Please keep these lines sorted alphabetically, as an automated tool (run every few months) is usually used to update this list.

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:137]: I know these weren't sorted before, but would you mind sorting them now, please?

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:204]: No need for this extra blank line, please remove.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Matthew-Reidy !
Just a couple minor tweaks, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by Matthew-Reidy** [comment]
Sure thing. I'll get I'll fix those items shortly

**Review by gmlewis — APPROVED** [review]
Thank you, @Matthew-Reidy !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

File: github/repos_rules.go
Comment (liaodaniel) [inline github/repos_rules.go:213]:
```suggestion
// NewMergeQueueRule creates a rule to only allow merges via a merge queue.
```

**Comment by liaodaniel** [comment]
Just a minor comment to update but looks good otherwise

**Comment by Matthew-Reidy** [comment]
changes to that comment have been made

**Review by liaodaniel — APPROVED** [review]
LGTM

**Pull request merged**
